### PR TITLE
[WIP]Run with mangata runtime 3000

### PR DESCRIPTION
**Pull request by imstar15**
### Test with mangata rococo-local

zombienet: https://github.com/OAK-Foundation/zombienet/tree/oak-fix

The task has been able to execute successfully on Turing and Mangata.

image

### Test with mangata-rococo

**Task on Turing**
https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.turing-staging.oak.tech#/explorer/query/0xf66140e39ff89870c8e80bf1af42884e871ee60ec4c05c8283f296958590820b

**XCM execution on Mangata**
https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fcollator-01-ws-rococo.mangata.online#/explorer/query/0x0b686dbb2138d1d42341a937620f567d8bacf9903f57e18e695646e93e63d47b

image

### CODE REVIEW

File: src/mangata/mangata.js [hunk 1]
```diff
@@ -69,8 +69,8 @@ async function main() {
 
     // TODO: determining liquidityTokenId by symbol name cannot handle duplicate symbols. It’s better we retrieve pools and find the correct pool
     const pools = await mangataHelper.getPools({ isPromoted: true });
-    const { liquidityTokenId } = _.find(pools, (pool) => pool.firstTokenId === mangataHelper.getTokenIdBySymbol(mangataNativeToken.symbol) && pool.secondTokenId === mangataHelper.getTokenIdBySymbol(turingNativeToken.symbol));
-
+    // const { liquidityTokenId } = _.find(pools, (pool) => pool.firstTokenId === mangataHelper.getTokenIdBySymbol(mangataNativeToken.symbol) && pool.secondTokenId === mangataHelper.getTokenIdBySymbol(turingNativeToken.symbol));
+    const liquidityTokenId = '8';
```
[the remark this answers is not in the document]
Comment (chrisli30) [reply src/mangata/mangata.js:73]: Remove the isPromoted: true in `mangataHelper.getPools()`

**Review by chrisli30 — APPROVED** [review]
The code looks okay. We don’t need to get reward in dev environment due to the change of updatePoolPromotion() for now.

**Review by chrisli30 — APPROVED** [review]
The code looks okay. We don’t need to get reward in dev environment due to the change of updatePoolPromotion() for now.

File: src/common/mangataHelper.js [hunk 2]
```diff
@@ -210,26 +210,23 @@ class MangataHelper {
...
-
-        const filterd = isPromoted ? _.filter(pools, (item) => item.isPromoted) : pools;
+        const filterdPools = isPromoted ? _.filter(pools, (item) => item.isPromoted) : pools;
+        return filterdPools;
+    };
 
-        const formatted = _.map(filterd, (item) => {
-            const firstToken = _.find(that.assets, { id: item.firstTokenId });
-            const firstTokenAmountFloat = (new BN(item.firstTokenAmount)).div(getDecimalBN(firstToken.decimals));
+    processPoolTokenAmountDecimals = (pool) => {
```
Comment (chrisli30) [inline src/common/mangataHelper.js:219]: This function is for `formatting`, as in the original variable name `formatted`. When you name the function you need to mean it. Do not use `process` for function name, as it’s too general.

**Review by chrisli30 — APPROVED** [review]

File: src/common/mangataHelper.js (hunk 2 above)
Comment (imstar15) [reply src/common/mangataHelper.js]: Fixed.

**Pull request merged**